os/arch/arm/src/imxrt/imxrt_wdog_lowerhalf.c

```diff
@@ -0,0 +1,264 @@
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <stdint.h>
+#include <errno.h>
+
+#include <tinyara/kmalloc.h>
+#include <tinyara/watchdog.h>
+#include <tinyara/irq.h>
+
+#include "imxrt_wdog.h"
+#include "chip/imxrt_wdog.h"
+
+#if defined(CONFIG_ARCH_CHIP_FAMILY_IMXRT102x)
+#include "chip/imxrt102x_config.h"
+#elif defined(CONFIG_ARCH_CHIP_FAMILY_IMXRT105x)
+#include "chip/imxrt105x_config.h"
+#else
+#error Unrecognized i.MX RT architecture
+#endif
+
+#ifdef CONFIG_WATCHDOG
+static int imxrt_wdog_start(FAR struct watchdog_lowerhalf_s *lower);
+static int imxrt_wdog_stop(FAR struct watchdog_lowerhalf_s *lower);
+static int imxrt_wdog_keepalive(FAR struct watchdog_lowerhalf_s *lower);
+static int imxrt_wdog_getstatus(FAR struct watchdog_lowerhalf_s *lower, FAR struct watchdog_status_s *status);
+static int imxrt_wdog_settimeout(FAR struct watchdog_lowerhalf_s *lower, uint32_t timeout);
+static xcpt_t imxrt_wdog_capture(FAR struct watchdog_lowerhalf_s *lower, xcpt_t handler);
+static int imxrt_wdog_ioctl(FAR struct watchdog_lowerhalf_s *lower, int cmd, unsigned long arg);
+
+const struct watchdog_ops_s g_wdgops = {
+	.start = imxrt_wdog_start,
+	.stop = imxrt_wdog_stop,
+	.keepalive = imxrt_wdog_keepalive,
+	.getstatus = imxrt_wdog_getstatus,
+	.settimeout = imxrt_wdog_settimeout,
+	.capture = imxrt_wdog_capture,
+	.ioctl = imxrt_wdog_ioctl,
+};
+
+/* This structure provides the private representation of the "lower-half"
+ * driver state structure.  This structure must be cast-compatible with the
+ * well-known watchdog_lowerhalf_s structure.
+ */
+
+struct imxrt_wdog_lowerhalf_s {
+	FAR const struct watchdog_ops_s *ops;
+	WDOG_Type *base;
+	wdog_config_t config;
+	xcpt_t handler;
+};
+
+static int imxrt_wdog_start(FAR struct watchdog_lowerhalf_s *lower)
+{
+	struct imxrt_wdog_lowerhalf_s *priv = (struct imxrt_wdog_lowerhalf_s *)lower;
+	DEBUGASSERT(priv);
+
+	imxrt_wdog_init(priv->base, &priv->config);
+	return OK;
+}
+
+static int imxrt_wdog_stop(FAR struct watchdog_lowerhalf_s *lower)
+{
+	struct imxrt_wdog_lowerhalf_s *priv = (struct imxrt_wdog_lowerhalf_s *)lower;
+	DEBUGASSERT(priv);
+
+	imxrt_wdog_deinit(priv->base);
+
+	return OK;
+}
+
+static int imxrt_wdog_keepalive(FAR struct watchdog_lowerhalf_s *lower)
+{
+	struct imxrt_wdog_lowerhalf_s *priv = (struct imxrt_wdog_lowerhalf_s *)lower;
+	DEBUGASSERT(priv);
+
+	imxrt_wdog_refresh(priv->base);
+
+	return OK;
+}
+
+static int imxrt_wdog_getstatus(FAR struct watchdog_lowerhalf_s *lower, FAR struct watchdog_status_s *status)
+{
+	int flag;
+	struct imxrt_wdog_lowerhalf_s *priv = (struct imxrt_wdog_lowerhalf_s *)lower;
+	DEBUGASSERT(priv);
+
+	flag = imxrt_wdog_getstatusflags(priv->base);
+
+	if ((flag & WDOG_WCR_WDE_MASK != 0) && (flag & WDOG_WCR_WDW_MASK == 0)) {
```

Comment:
If user just call as a below, it will return WDFLAGS_ACTIVE;
open();
ioctl(fd, WDIOC_GESTATUS);
So, some default should be set, I think WDFLAG_RESET (timer expried)

Reply:
Done. Thanks.